packages/web3-core/src/web3_request_manager.ts

```diff
@@ -64,22 +70,22 @@ export class Web3RequestManager extends Web3EventEmitter<{
 		return this._providers;
 	}
 
-	public setProvider(provider: SupportedProviders | string, net?: Socket) {
-		let newProvider!: Web3BaseProvider;
+	public setProvider(provider: SupportedProviders<API> | string, net?: Socket) {
```

Comment:
added issue for validating current design choices https://github.com/ChainSafe/web3.js/issues/4543

Reply:
We need integration tests for every thing, not just this part.